specs/APIReview_TextureStream.md

```diff
@@ -0,0 +1,353 @@
+	
+# Background
+Many native apps use a native engine for real-time communication scenarios, which include video 
+capture, networking and video rendering.  However, often, these apps still use WebView or 
+Electrion for UI rendering. The separation between real-time video rendering and UI rendering 
+prevents apps from rendering real-time video inside the web contents. This forces apps to
+render the real-time video on top of the web contents, which is limiting. Rendering video on 
+top constrains the user experience and it may also cause performance problems.  
+We can ask the native apps to use web renderer for video handling because web standard already 
+provides these features through WebRTC APIs. The end developers, however, prefer to use 
+their existing engine such as capturing and composition, meanwhile using WebRTC API for rendering.
+
+# Description
+The proposed APIs will allow the end developers to stream the captured or composed video frame to 
+the WebView renderer where Javascript is able to insert the frame to the page through W3C standard 
+API of Video, MediaStream element for displaying it.
+The API will use the shared GPU texture buffer so that it can minimize the overall cost with 
+regards to frame copy.
+
+# Examples
+Javascript
+// User click the video capture button.
+document.querySelector('#showVideo').addEventListener('click',
+  e => getStreamFromTheHost(e));
+async function getStreamFromTheHost(e) {
+  try {
+    // Request stream to the host with unique stream id.
+    const stream = await window.chrome.webview.getTextureStream('webview2-abcd1234'); 
```

Comment:
Is the only scenario for this API to have native sending video to JS via MediaStream to render in web content? Maybe I'm thinking of a different scenario where the JS side could edit or create video to send back to native.

Reply:
getTextureStream returns W3C standard of MediaStream/MediaStreamTtrack objects and developer can call commands/events on them as documented on W3C spec. 

Yes, one of things they can do with these objects is to edit (insertable) the video frame. However, it is also W3C standard, no special things for this API.

Do you think we need to add it even though there is no difference (https://web.dev/mediastreamtrack-insertable-media-processing/)?

Reply:
Insertable stream API (W3C standard API) is the one that allow editing texture stream, but it is used for another API of 'registerTextureStream', which will come on next round. (It is under design review, and shortly on PR in this week).